providers/desec/protocol.go

```diff
@@ -71,37 +73,138 @@ func (c *desecProvider) authenticate() error {
 	}
 	return nil
 }
+func (c *desecProvider) initializeDomainIndex() error {
+	if c.domainIndexInitialized {
```

Comment:
Suggestion: Protect domainIndex with a mutex.  It isn't needed now, but some day we'll call these concurrently.

Reply:
Done